validator/src/main/java/com/google/daq/mqtt/mapping/MappingAgent.java

```diff
@@ -141,25 +144,66 @@ private FamilyDiscoveryConfig getFamilyDiscoveryConfig(String family) {
     return familyDiscoveryConfig;
   }
 
-  private void reconcileDiscovery() {
+  private void mapDiscoveredDevices() {
+    List<Entry<String, Metadata>> mappedDiscoveredEntries = getMappedDiscoveredEntries();
+    Map<String, Metadata> devicesEntriesMap = getDevicesEntries();
+
+    mappedDiscoveredEntries.forEach(entry -> {
+
+      if (devicesEntriesMap.containsKey(entry.getKey())) {
+        System.err.println("Skipping existing device file for family::address = " + entry.getKey());
+        //TODO: update the existing device
+      } else {
+        // UNK denotes unknown device
+        String newDeviceName = "UNK-" + suffixToStart;
+        suffixToStart++;
```

Comment:
I think you might need something a bit more sophisticated here... e.g. need to cehck the devicesEntryMap to find the next not-used UNK- entry, otherwise this might clobber existing UNK- devices.